cico_deploy.sh

```diff
@@ -2,69 +2,88 @@
 set -u
 set +e
 
+echo ""
+echo "#####################################"
+echo "     STARTING BUILD VALIDATION"
+echo "#####################################"
+echo ""
+
 # Source build variables
 cat jenkins-env | grep PASS > inherit-env
 . inherit-env
 . config 
 . ~/che_image_tag.env
 
+echo "VALIDATION: Validating image ${DOCKER_HUB_NAMESPACE}/che-server:${CHE_SERVER_DOCKER_IMAGE_TAG} =="
+
 # Install oc client
 yum install -y centos-release-openshift-origin
 yum install -y origin-clients
 
+OSO_USER=mloriedo
 OSO_TOKEN=$(curl -sSL -H "X-Vault-Token: ${DOCKER_HUB_PASSWORD}" -H "Content-Type: application/json" \
 	-X GET http://li546-232.members.linode.com:8200/v1/secret/os_token | cut -d\" -f18)
 if [[ -z "$OSO_TOKEN" ]]; then
-    echo "OSO token is empty, cannot proceed with verification"
+    echo "VALIDATION: Error OSO token is empty, cannot proceed with verification"
     exit 1
 fi
-OSO_USER=mloriedo
+echo "VALIDATION: Retrieved OSO token for user ${OSO_USER}"
+
+OSIO_VERSION=$(curl -sSL http://central.maven.org/maven2/io/fabric8/online/apps/che/maven-metadata.xml | grep latest | sed -e 's,.*<latest>\([^<]*\)</latest>.*,\1,g')
+echo "VALIDATION: Retrieved latest OSIO version to use for deployment: ${OSIO_VERSION}"
+
 OSO_DOMAIN=8a09.starter-us-east-2.openshiftapps.com
 OSO_API_ENDPOINT=https://api.starter-us-east-2.openshift.com
-OSIO_VERSION=$(curl -sSL http://central.maven.org/maven2/io/fabric8/online/apps/che/maven-metadata.xml | grep latest | sed -e 's,.*<latest>\([^<]*\)</latest>.*,\1,g')
 CHE_OPENSHIFT_HOSTNAME=${OSO_USER}-che.${OSO_DOMAIN}
 CHE_OPENSHIFT_PROJECT=${OSO_USER}-che
 
 oc login ${OSO_API_ENDPOINT} --token=${OSO_TOKEN}
 oc project ${CHE_OPENSHIFT_PROJECT}
-echo "Getting version of OSIO and applying template"
+
+echo "VALIDATION: Cleaning-up previous deployment"
+oc delete all -l project=che
+
+echo "VALIDATION: Deploying che-server"
 curl -sSL http://central.maven.org/maven2/io/fabric8/online/apps/che/${OSIO_VERSION}/che-${OSIO_VERSION}-openshift.yml | \
     sed "s/    hostname-http:.*/    hostname-http: ${CHE_OPENSHIFT_HOSTNAME}/" | \
     sed "s/          image:.*/          image: ${DOCKER_HUB_NAMESPACE}\/che-server:${CHE_SERVER_DOCKER_IMAGE_TAG}/" | \
```

Comment:
I think this is the real problem, because it uses RH_CHE_DOCKER_HUB_NAMESPACE var and that one points to docker.io/rhchestage. That mean we have one more slash in sed command and that is causing problem. See log of latest failing job https://ci.centos.org/view/Devtools/job/devtools-build-run-che-build-master/322/console : 

```sed: -e expression #1, char 49: unknown option to `s'```
`error: no objects passed to apply`
`No resources found.`
`Checking state of Che pod.`